scripts/provision_health_checks.py

```diff
@@ -51,30 +52,22 @@ def get_health_checks():
     return dcp_health_checks
 
 
-def health_check_ops():
-    client = boto3.client('route53')
-    paginator = client.get_paginator('list_health_checks')
-    return client, chain.from_iterable(page['HealthChecks'] for page in paginator.paginate())
-
-
 def provision_health_check(health_checks, dcp_health_check, reference):
-    dcp_health_check_config = dcp_health_check['HealthCheckConfig']
-    azul_service = health_checks[config.service_name]
-    azul_indexer = health_checks[config.indexer_name]
-    azul_data_browser = health_checks[config.data_browser_name]
-    azul_data_portal = health_checks[config.data_portal_name]
-    updated_child_health_checks = [health_check for health_check in dcp_health_check_config['ChildHealthChecks']
-                                   if health_check not in (azul_service['Id'],
-                                                           azul_indexer['Id'],
-                                                           azul_data_browser['Id'],
-                                                           azul_data_portal['Id'])]
-    if reference:
-        updated_child_health_checks.append(azul_service['Id'])
-        updated_child_health_checks.append(azul_indexer['Id'])
-        updated_child_health_checks.append(azul_data_browser['Id'])
-        updated_child_health_checks.append(azul_data_portal['Id'])
-    response = update_dcp_wide_health_check(dcp_health_check, updated_child_health_checks)
-    logger.info(json.dumps(response))
+    dcp_health_check_children = dcp_health_check['HealthCheckConfig']['ChildHealthChecks']
+    try:
+        azul_health_checks = [health_checks[resource_name]['Id'] for resource_name in (config.service_name,
+                                                                                       config.indexer_name,
+                                                                                       config.data_browser_name,
+                                                                                       config.data_portal_name)]
+    except KeyError:
+        logger.info(f'No Azul health checks found for {config.deployment_stage} deployment!')
```

Comment:
It's technically possible that some of the health checks are not provisioned, but the others are. Consider if you want to explicitly handle this situation.